src/axom/quest/tests/quest_pro_e_reader.cpp

```diff
@@ -0,0 +1,299 @@
+  namespace fs = axom::utilities::filesystem;
+  std::string cup = fs::joinPath(AXOM_DATA_DIR, "quest/cup.proe");
+
+  // STEP 1: create a Pro/E reader and read-in the mesh data
+  axom::quest::ProEReader reader;
+  reader.setFileName(cup);
+  int status = reader.read();
+  EXPECT_EQ(status, 0);
+
+  // STEP 2: reading the Pro/E mesh data into a axom::mint::Mesh
+  axom::mint::UnstructuredMesh<axom::mint::SINGLE_SHAPE> mesh(3, axom::mint::TET);
+  reader.getMesh(&mesh);
+
+  // STEP 3: ensure the mesh is what is expected
+  EXPECT_EQ(mesh.getNumberOfCells(), NUM_TETS);
+  EXPECT_EQ(mesh.getNumberOfNodes(), NUM_TETS * 4);
+
+  const double* x = mesh.getCoordinateArray(axom::mint::X_COORDINATE);
+  const double* y = mesh.getCoordinateArray(axom::mint::Y_COORDINATE);
+  const double* z = mesh.getCoordinateArray(axom::mint::Z_COORDINATE);
+  EXPECT_TRUE(x != nullptr);
+  EXPECT_TRUE(y != nullptr);
+  EXPECT_TRUE(z != nullptr);
+
+  // STEP 4: Verify a few tetrahedra are as expected
+
+  // Check first tetrahedron
+  EXPECT_NEAR(x[0], -71.842646, EPS);
+  EXPECT_NEAR(x[1], 18.091620, EPS);
+  EXPECT_NEAR(x[2], 18.091620, EPS);
+  EXPECT_NEAR(x[3], 16.637112, EPS);
+
+  EXPECT_NEAR(y[0], -142.432949, EPS);
+  EXPECT_NEAR(y[1], -100.389772, EPS);
+  EXPECT_NEAR(y[2], -158.496833, EPS);
+  EXPECT_NEAR(y[3], -116.282419, EPS);
+
+  EXPECT_NEAR(z[0], 0.000000, EPS);
+  EXPECT_NEAR(z[1], 0.000000, EPS);
+  EXPECT_NEAR(z[2], 0.000000, EPS);
+  EXPECT_NEAR(z[3], -53.197840, EPS);
+
+  // Check last tetrahedron
+  constexpr int LAST_OFFSET = (NUM_TETS - 1) * 4;
+
+  EXPECT_NEAR(x[LAST_OFFSET], 128.940529, EPS);
+  EXPECT_NEAR(x[LAST_OFFSET + 1], 68.261330, EPS);
+  EXPECT_NEAR(x[LAST_OFFSET + 2], 121.442679, EPS);
+  EXPECT_NEAR(x[LAST_OFFSET + 3], 68.137155, EPS);
+
+  EXPECT_NEAR(y[LAST_OFFSET], -93.930323, EPS);
+  EXPECT_NEAR(y[LAST_OFFSET + 1], -41.261753, EPS);
+  EXPECT_NEAR(y[LAST_OFFSET + 2], -31.346653, EPS);
+  EXPECT_NEAR(y[LAST_OFFSET + 3], -41.466486, EPS);
+
+  EXPECT_NEAR(z[LAST_OFFSET], -145.985725, EPS);
+  EXPECT_NEAR(z[LAST_OFFSET + 1], -122.672539, EPS);
+  EXPECT_NEAR(z[LAST_OFFSET + 2], -174.327849, EPS);
+  EXPECT_NEAR(z[LAST_OFFSET + 3], -173.363017, EPS);
+
+  // Check middle tetrahedron
+  constexpr int MID_OFFSET = ((NUM_TETS / 2) - 1) * 4;
+
+  EXPECT_NEAR(x[MID_OFFSET], 15.508351, EPS);
+  EXPECT_NEAR(x[MID_OFFSET + 1], -30.194340, EPS);
+  EXPECT_NEAR(x[MID_OFFSET + 2], -35.211560, EPS);
+  EXPECT_NEAR(x[MID_OFFSET + 3], 14.765131, EPS);
+
+  EXPECT_NEAR(y[MID_OFFSET], 124.048744, EPS);
+  EXPECT_NEAR(y[MID_OFFSET + 1], 109.246291, EPS);
+  EXPECT_NEAR(y[MID_OFFSET + 2], 117.715966, EPS);
+  EXPECT_NEAR(y[MID_OFFSET + 3], 78.384516, EPS);
+
+  EXPECT_NEAR(z[MID_OFFSET], -179.442905, EPS);
+  EXPECT_NEAR(z[MID_OFFSET + 1], -112.054428, EPS);
+  EXPECT_NEAR(z[MID_OFFSET + 2], -178.748301, EPS);
+  EXPECT_NEAR(z[MID_OFFSET + 3], -163.120569, EPS);
+
+  // Step 5: Dump mesh file
+  axom::mint::write_vtk(&mesh, "cup.vtk");
```

Comment:
Is writing the vtk sufficiently fast? Should it be optional/ commented out by default?

Reply:
> Is writing the vtk sufficiently fast? Should it be optional/ commented out by default?

It's fast in my experience (~0.30 seconds for all the unit tests), so I think it is fine if enabled by default.

I did notice there are a few Axom unit tests writing out meshes and other files, some optional and others not.
There does not look to be a standard way of enabling/disabling all these output options(commented out, #ifdef out, etc.).
Maybe something useful to standardize?